packages/ethereum-contracts/test/foundry/utils/TOGA.t.sol

```diff
@@ -0,0 +1,553 @@
+        vm.assume(bond > 1<<32);
+
+        _sendPICBid(alice, superToken, bond, EXIT_RATE_1E3);
+        _assetNetflow(superToken, alice, EXIT_RATE_1E3);
+
+        vm.warp(block.timestamp + 1000);
+        _deleteFlow(superToken, address(toga), alice);
+        _assetNetflow(superToken, alice, 0);
+
+        vm.startPrank(alice);
+        toga.changeExitRate(superToken, 0);
+        _assetNetflow(superToken, alice, 0);
+
+        toga.changeExitRate(superToken, EXIT_RATE_1);
+        _assetNetflow(superToken, alice, EXIT_RATE_1);
+
+        vm.stopPrank();
+
+        // stop again and let bob make a bid
+        vm.warp(block.timestamp + 1000);
+        _deleteFlow(superToken, address(toga), alice);
+        _assetNetflow(superToken, alice, 0);
+
+        _sendPICBid(bob, superToken, bond, EXIT_RATE_1E3);
+        _assetNetflow(superToken, alice, 0);
+        _assetNetflow(superToken, bob, EXIT_RATE_1E3);
+    }
+
+    function testCollectedRewardsAreAddedToThePICBond(uint256 bond, uint256 rewards) public {
+        vm.assume(bond > 0);
+        vm.assume(bond < INIT_SUPER_TOKEN_BALANCE); // User only has 64 bits test super tokens
+        vm.assume(bond > 1<<32);
+        vm.assume(rewards > 0);
+        vm.assume(rewards < INIT_SUPER_TOKEN_BALANCE);
+
+        _sendPICBid(alice, superToken, bond, 0);
+
+        _transferToken(superToken, admin, address(toga), rewards);
+
+        (, uint256 picBond, ) = toga.getCurrentPICInfo(superToken);
+        assertTrue(picBond == bond + rewards);
+    }
+
+    function testBondIsConsumedByExitFlow(uint256 aliceBond) public {
+        vm.assume(aliceBond > 0);
+        vm.assume(aliceBond < INIT_SUPER_TOKEN_BALANCE / 2); // User only has type(uint64).max test super tokens
+        vm.assume(aliceBond > 1<<32);
+
+        int96 maxRate = _shouldMaxExitRate(aliceBond);
+        _sendPICBid(alice, superToken, aliceBond, maxRate);
+        _assetNetflow(superToken, alice, maxRate);
+
+        // critical stream is liquidated - remaining bond goes to zero
+        vm.warp(block.timestamp + 1e6);
+
+        // A sentinel would do this
+        vm.startPrank(admin);
+        superToken.deleteFlow(address(toga), alice);
+        vm.stopPrank();
+
+        _assetNetflow(superToken, alice, 0);
+        _assetNetflow(superToken, address(toga), 0);
+
+        // this assumes the flow deletion was not triggered by the PIC - otherwise rewards would be accrued
+        (, uint256 bond, ) = toga.getCurrentPICInfo(superToken);
+        assertTrue(bond == 0);
+
+        // alice tries to re-establish stream - fail because no bond left
+        vm.startPrank(alice);
+        vm.expectRevert("TOGA: exitRate too high");
+        toga.changeExitRate(superToken, _shouldDefaultExitRate(aliceBond));
+        vm.stopPrank();
+
+        deal(address(superToken), address(toga), 1e15);
+
+        (, uint256 bond2, ) = toga.getCurrentPICInfo(superToken);
+        assertTrue(bond2 >= 1e12);
+
+        vm.startPrank(alice);
```

Comment:
the changing of the exit rate likely can be placed in a helper function where you assertNetFlow after.